src/main/java/com/connexta/store/adaptors/StorageAdaptor.java

```diff
@@ -39,4 +39,12 @@ void store(
    */
   @NotNull
   StorageAdaptorRetrieveResponse retrieve(@NotBlank final String key) throws RetrieveException;
+
+  /**
+   * Updates the {@link StoreStatus} of {@code dataset} for a given {@code datasetId}
+   *
+   * @param datasetId the Id of a dataset
+   * @param storeStatus the new {@link StoreStatus} for a dataset
+   */
+  void updateStatus(String datasetId, String storeStatus);
```

Comment:
❗️Add validation constraints on datasetId and storeStatus (and tests)

Reply:
I will, but I'm still not convinced that they do anything. (If you're referring to annotations).

Reply:
If annotations work, that's good. But we should protect against null and empty values at least.